backend.native/tests/interop/objc/tests/kt41811.kt

```diff
@@ -81,3 +81,29 @@ private fun createGarbageDeallocLoadWeak(weakRef: WeakReferenceProtocol) {
         assertSame(obj, weakDeallocLoadWeak!!.referent)
     }
 }
+
+@Test
+fun testKT41811WithGlobal() {
+    // Attempt to make the state predictable:
+    kotlin.native.internal.GC.collect()
+
+    deallocRetainReleaseDeallocated = false
+    assertFalse(deallocRetainReleaseDeallocated)
+
+    autoreleasepool {
+        {
+            globalDeallocRetainRelease = object: DeallocRetainRelease() {}
+        }()
+        // Clean up local DeallocRetainRelease on Kotlin side
+        kotlin.native.internal.GC.collect()
+        // And drop the last reference to DeallocRetainRelease from ObjC global scope.
+        globalDeallocRetainRelease = null
+    }
+
+    assertFalse(deallocRetainReleaseDeallocated)
+
+    // This will dispose DeallocRetainRelease on Kotlin side, which will cause dealloc
+    // on ObjC side, which will immediately schedule another disposal on the Kotlin side.
```

Comment:
The comment should be more conditional: mention that this happens if the fix isn't applied.
See e.g. "if the dealloc applied addRef/releaseRef to reclaimed Kotlin object" above.